### ISSUE TITLE
Add feature to capture OPL events to .VGM format.

### ISSUE THREAD

**Issue by Corak667** [joncampbell123/dosbox-x#4504]
### Is your feature request related to a problem? Please describe.

Hi there. 
There still no support for .vgm format.
It's extremely popular in vgmrips team and they still using old official dosbox with VGM mod:
https://vgmrips.net/
Your dosbox-x have VERY useful features for vgm rippers:
1. Savestates.
2. Pause with opened interrups.

### What you want

Please add feature to capture OPL events to .VGM format. 
It currently more actural for vgm rippers and have much more utilites and chips supported.

### Describe alternatives you've considered

There is already VGM mod from vgmrips team by NewRisingSun2 and ValleyBell:
https://vgmrips.net/forum/viewtopic.php?f=3&t=2585&hilit=Dosbox
Latest build from 2017:
https://vgmrips.net/forum/download/file.php?id=481
src diffs:
https://vgmrips.net/forum/download/file.php?id=480

### Additional information

[dosbox-r4000-vgmcapture.zip](https://github.com/joncampbell123/dosbox-x/files/12795084/dosbox-r4000-vgmcapture.zip)
[DOSBox-VGM build.zip](https://github.com/joncampbell123/dosbox-x/files/12795085/DOSBox-VGM.build.zip)

### Have you checked that no similar feature request(s) exist?

- [X] I have searched and didn't find any similar feature request.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by zhblue** [comment]
Yes! I have been switching between dosbox-x and dosbox-vgm quite a while, if the capture capability is integrated into dosbox , there will be a lot more fun.

**Comment by joncampbell123** [comment]
Speaking of patches, any interesting the "multi track audio capture"? Which is basically WAV capture except audio is written to an AVI and each mixer audio channel is given it's own stream in the AVI file. The intent is that you can use it to record in a way that you can easily separate the FM synth from the Sound Blaster output if needed in video editing, for example.

**Comment by joncampbell123** [comment]
https://vgmrips.net/wiki/VGM_Specification

**Comment by joncampbell123** [comment]
I wonder if the VGM format could be expanded to record GF1 Gravis Ultrasound events too, since there seems to be consideration of PCM sample uploads in the VGM format.

**Comment by joncampbell123** [comment]
> Yes! I have been switching between dosbox-x and dosbox-vgm quite a while, if the capture capability is integrated into dosbox , there will be a lot more fun.

I'm for it. Do you need me to do it or do you plan on making a pull request?

**Comment by johnnovak** [comment]
> Speaking of patches, any interest in the "multi track audio capture"? Which is basically WAV capture except audio is written to an AVI and each mixer audio channel is given it's own stream in the AVI file.

Just noting that the WAV format does support arbitrary number of channels, you don't need AVI for that. In fact, AVI is worse I think because then you can not use standard audio editing tools easily or at all to edit the multichannel data (such as REAPER and SoundForge).

**Comment by Torinde** [comment]
@Corak667, you put thumbs up on Jon's question, but what is your answer:
1. Do you plan on making a pull request?
2. Do you need Jon to do it?

**Comment by Corak667** [comment]
1,
2 - Jon is spec in Dosbos-X, so he will do it bettter
